dcicutils/snapshot_utils.py

```diff
@@ -0,0 +1,416 @@
+                                       repository_short_name=repository_short_name,
+                                       indexer_namespace=indexer_namespace, snapshot_name=snapshot_name)
+            repo_creation_result = es.snapshot.create_repository(repository_short_name, spec)
+            assert repo_creation_result == {'acknowledged': True}
+            if cls.VERBOSE_SNAPSHOTS:
+                PRINT("Creating snapshot", repository_short_name)
+            ignored(indexer_namespace)  # It would be nice to makea  snapshot only of part of the data
+            snapshot_creation_result = es.snapshot.create(repository=repository_short_name,
+                                                          snapshot=snapshot_name,
+                                                          wait_for_completion=True)
+            assert snapshot_creation_result.get('snapshot', {}).get('snapshot') == snapshot_name
+        except Exception as e:
+            logging.error(str(e))
+            if cls.DEBUG_SNAPSHOTS:
+                import pdb
+                pdb.set_trace()
+            raise
+
+    @classmethod
+    def restore_snapshot(cls, es_testapp, indexer_namespace, repository_short_name, snapshot_name=None,
+                         require_indexer_namespace=True):
+        if not cls.ENABLE_SNAPSHOTS:
+            if cls.VERBOSE_SNAPSHOTS:
+                PRINT("NOT restoring snapshot because %s.ENABLE_SNAPSHOTS is False." % full_object_name(cls))
+            return
+        es = es_testapp.app.registry['elasticsearch']
+        try:
+            if require_indexer_namespace:
+                if not indexer_namespace or not cls.is_valid_indexer_namespace(indexer_namespace):
+                    raise RuntimeError("restore_snapshot requires an indexer namespace prefix (got %r)."
+                                       " (You can use the indexer_namespace fixture to acquire it.)"
+                                       % (indexer_namespace,))
+
+            all_index_info = [info['index'] for info in es.cat.indices(format='json')]
+            index_names = [name for name in all_index_info if name.startswith(indexer_namespace)]
+            if index_names:
+                index_names_string = ",".join(index_names)
+                if cls.VERBOSE_SNAPSHOTS:
+                    # PRINT("Deleting index files", index_names_string)
+                    PRINT("Deleting index files for prefix=", indexer_namespace)
+                result = es.indices.delete(index_names_string)
+                if cls.VERBOSE_SNAPSHOTS:
+                    ignorable(result)
+                    # PRINT("deletion result=", result)
+                    PRINT("Deleted index files for prefix=", indexer_namespace)
+            result = es.snapshot.restore(repository=repository_short_name,
+                                         snapshot=snapshot_name,
+                                         wait_for_completion=True)
```

Comment:
Docstring would clarify the expected behavior here (you are wiping all prefixed indices prior to a restore - you don't necessarily need/want to do that).

Moreover docstring is missing on a few of these (not marking), but those that do have it are documented well.

Reply:
It doesn't like to restore when there is stuff already there.